src/test/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjectionTest.java

```diff
@@ -0,0 +1,119 @@
+        verify(commandInjection).getResponseFromPingCommand(eq(LOCALHOST), eq(true));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_NoIp_ExpectNoResponseContent() throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(Strings.EMPTY);
+
+        // Assert
+        assertStatusAndValid(responseEntity);
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(Strings.EMPTY);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(eq(Strings.EMPTY));
+        verify(commandInjection).getResponseFromPingCommand(eq(Strings.EMPTY), eq(false));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_Localhost_ExpectPingSuccess()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(LOCALHOST));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(LOCALHOST, entity);
+
+        assertStatusAndValid(responseEntity);
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(successReturnStr);
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(LOCALHOST), eq(entity));
+        verify(commandInjection).getResponseFromPingCommand(eq(LOCALHOST), eq(true));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_NoIp_ExpectNoResponseContent()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(Strings.EMPTY));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(Strings.EMPTY, entity);
+
+        // Assert
+        assertStatusAndValid(responseEntity);
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(Strings.EMPTY);
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(Strings.EMPTY), eq(entity));
+        verify(commandInjection).getResponseFromPingCommand(eq(Strings.EMPTY), eq(false));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_LocalhostAndSEMICOLON_ExpectNoResponseContent()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity =
+                new RequestEntity(HttpMethod.GET, URI.create("http://localhost:8080/;"));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(LOCALHOST, entity);
+
+        // Assert
+        assertStatusAndValid(responseEntity);
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(Strings.EMPTY);
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(LOCALHOST), eq(entity));
+        verify(commandInjection).getResponseFromPingCommand(eq(LOCALHOST), eq(false));
```

Comment:
These 5 lines can be wrapped inside a method and based on condition as false, responseEntity.getBody().getContent() is empty and based on condition as true response is successStr.

Reply:
so that this common code can be used at all the tests.